Panic runing config  validate

Panic runing config  validate

Build using yay -S fan2go-git
run: fan2go config validate

[ed@grover fan2go]$ sudo fan2go config validate
  FATAL   Error reading config file, While parsing config: yaml: line 30: did not find expected '-' indicator
panic: 

goroutine 1 [running]:
github.com/pterm/pterm.checkFatal(...)
        github.com/pterm/pterm@v0.12.65/prefix_printer.go:351
github.com/pterm/pterm.(*PrefixPrinter).Printfln(0x55d6ad0eade0, {0x55d6acab1948, 0x1d}, {0xc0000f3c88, 0x1, 0x1})
        github.com/pterm/pterm@v0.12.65/prefix_printer.go:293 +0x1ed
github.com/markusressel/fan2go/internal/ui.Fatal({0x55d6acab1948, 0x1d}, {0xc0000f3c88, 0x1, 0x1})
        github.com/markusressel/fan2go/internal/ui/logging.go:52 +0x8b
github.com/markusressel/fan2go/internal/configuration.ReadInConfig()
        github.com/markusressel/fan2go/internal/configuration/config.go:111 +0x5e
github.com/markusressel/fan2go/internal/configuration.DetectAndReadConfigFile(...)
        github.com/markusressel/fan2go/internal/configuration/config.go:103
github.com/markusressel/fan2go/cmd/config.glob..func1(0x55d6ad0ee900?, {0xc00021a350?, 0x1?, 0x1?})
        github.com/markusressel/fan2go/cmd/config/validate.go:16 +0x1e
github.com/spf13/cobra.(*Command).execute(0x55d6ad0ee900, {0xc00021a330, 0x1, 0x1})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x55d6ad0ee060)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.7.0/command.go:992
github.com/markusressel/fan2go/cmd.Execute()
        github.com/markusressel/fan2go/cmd/root.go:88 +0x115
main.main()
        ./main.go:25 +0x17
[ed@grover fan2go]$

**Expected behavior**
Errors in the config should be displayed correctly, there should not be a panic.

**Desktop (please complete the following information):**

- Distro: Arch Linux
- uname -a:  Linux grover 6.4.10-1-stable-git-00055-gdbf67f639df8-dirty #1 SMP PREEMPT_DYNAMIC
                      kernel built with the 6.5 hwmon pull to support nct6799-isa-0290
- sensors -v: sensors version 3.6.0+git with libsensors version 3.6.0+git
- fan2go version: dev (0.8.0.r151.g7c86ff0-1)

Add any other context about the problem here.

I've tried go-beta (same errors as above) an gcc-go (gets a compile error).  
My fan2go.yaml and output of detect included in setup.zip

[setup.zip](https://github.com/markusressel/fan2go/files/12363300/setup.zip)

--- a/cmd/curve/list.go
+++ b/cmd/curve/list.go
@@ -26,7 +26,7 @@ var curveCmd = &cobra.Command{
 
 		err = configuration.Validate(configPath)
 		if err != nil {
-			ui.Fatal(err.Error())
+			ui.FatalWithoutStacktrace(err.Error())
 		}
 
 		curveConfigsToPrint := []configuration.CurveConfig{}

--- a/cmd/fan/curve.go
+++ b/cmd/fan/curve.go
@@ -24,7 +24,7 @@ var curveCmd = &cobra.Command{
 		configuration.LoadConfig()
 		err := configuration.Validate(configPath)
 		if err != nil {
-			ui.Fatal(err.Error())
+			ui.FatalWithoutStacktrace(err.Error())
 		}
 
 		persistence := persistence.NewPersistence(configuration.CurrentConfig.DbPath)

--- a/cmd/fan/fan.go
+++ b/cmd/fan/fan.go
@@ -35,7 +35,7 @@ func getFan(id string) (fans.Fan, error) {
 	configuration.LoadConfig()
 	err := configuration.Validate(configPath)
 	if err != nil {
-		ui.Fatal(err.Error())
+		ui.FatalWithoutStacktrace(err.Error())
 	}
 
 	controllers := hwmon.GetChips()

--- a/cmd/sensor/sensor.go
+++ b/cmd/sensor/sensor.go
@@ -52,7 +52,7 @@ func getSensor(id string) (sensors.Sensor, error) {
 	configuration.LoadConfig()
 	err := configuration.Validate(configPath)
 	if err != nil {
-		ui.Fatal(err.Error())
+		ui.FatalWithoutStacktrace(err.Error())
 	}
 
 	controllers := hwmon.GetChips()

--- a/internal/backend.go
+++ b/internal/backend.go
@@ -143,7 +143,7 @@ func RunDaemon() {
 		}
 
 		if len(fans.FanMap) == 0 {
-			ui.Fatal("No valid fan configurations, exiting.")
+			ui.FatalWithoutStacktrace("No valid fan configurations, exiting.")
 		}
 	}
 	{

--- a/internal/configuration/config.go
+++ b/internal/configuration/config.go
@@ -100,19 +100,19 @@ func setDefaultValues() {
 
 // DetectAndReadConfigFile detects the path of the first existing config file
 func DetectAndReadConfigFile() string {
-	ReadInConfig()
+	err := readInConfig()
+	if err != nil {
+		ui.FatalWithoutStacktrace("Error reading config file, %s", err)
+	}
 	return GetFilePath()
 }
 
-// ReadInConfig reads and parses the config file
-func ReadInConfig() {
-	if err := viper.ReadInConfig(); err != nil {
-		// config file is required, so we fail here
-		ui.Fatal("Error reading config file, %s", err)
-	}
+// readInConfig reads and parses the config file
+func readInConfig() error {
+	return viper.ReadInConfig()
 }
 
-// GetFilePath this is only populated _after_ ReadInConfig()
+// GetFilePath this is only populated _after_ readInConfig()
 func GetFilePath() string {
 	return viper.ConfigFileUsed()
 }

--- a/internal/ui/logging.go
+++ b/internal/ui/logging.go
@@ -3,6 +3,7 @@ package ui
 import (
 	"fmt"
 	"github.com/pterm/pterm"
+	"os"
 )
 
 func SetDebugEnabled(enabled bool) {
@@ -45,6 +46,12 @@ func Error(format string, a ...interface{}) {
 func ErrorAndNotify(title string, format string, a ...interface{}) {
 	Error(format, a...)
 	NotifyError(title, fmt.Sprintf(format, a...))
+}
+
+func FatalWithoutStacktrace(format string, a ...interface{}) {
+	NotifyError("Fatal Error", fmt.Sprintf(format, a...))
+	pterm.Fatal.WithFatal(false).Printfln(format, a...)
+	os.Exit(1)
 }
 
 func Fatal(format string, a ...interface{}) {

